### PR TITLE
[Karazhan] rework - part 2

### PR DESCRIPTION
**Pull request by miraco**
Correct xyz coords for all trash mobs linked to midnight (attumen the horseman) Remove creature_linking and add all creatures into spawn_group system, this also fixes pathing of 1 patrol. remove ooc rp for some mobs (got handled via dbscripts_on_movement (will get handled via guid based eai)

Ill let this as draft for now because it also breaks something that worked with linking before.

Bugged atm:
1. When players attack midnight, all trashs around him should aggro

### CODE REVIEW

File: Updates/Instances/532_karazhan.sql [hunk 1]
```diff
@@ -3396,7 +3324,14 @@ INSERT INTO `spawn_group_entry` (`Id`, `Entry`, `MinCount`, `MaxCount`, `Chance`
...
 (@SGGUID+24, 181845, 0, 0, 0), (@SGGUID+24, 181846, 0, 0, 0), (@SGGUID+24, 181847, 0, 0, 0), (@SGGUID+24, 181848, 0, 0, 0), (@SGGUID+24, 181768, 0, 0, 0), -- Sealed Tome (Loot)
-(@SGGUID+24, 181833, 0, 0, 2), (@SGGUID+24, 181837, 0, 0, 2), (@SGGUID+24, 181839, 0, 0, 2), (@SGGUID+24, 181841, 0, 0, 2), (@SGGUID+24, 181843, 0, 0, 2); -- Sealed Tome (Trap)
+(@SGGUID+24, 181833, 0, 0, 2), (@SGGUID+24, 181837, 0, 0, 2), (@SGGUID+24, 181839, 0, 0, 2), (@SGGUID+24, 181841, 0, 0, 2), (@SGGUID+24, 181843, 0, 0, 2), -- Sealed Tome (Trap)
+-- Creatures
+(@SGGUID+50, 15547, 0, 0, 0), (@SGGUID+50, 15548, 0, 0, 0), -- Spectral Charger, Spectral Stallion
+(@SGGUID+52, 15547, 0, 0, 0), (@SGGUID+52, 15548, 0, 0, 0); -- Spectral Charger, Spectral Stallion
+
+INSERT INTO `spawn_group_linked_group` (`Id`, `LinkedId`) VALUES 
+-- Midnight linked trash, attack when midnight gets attacked. 
+(@SGGUID+25, @SGGUID+50), (@SGGUID+25, @SGGUID+51), (@SGGUID+25, @SGGUID+52), (@SGGUID+25, @SGGUID+53),(@SGGUID+25, @SGGUID+54), (@SGGUID+25, @SGGUID+55), (@SGGUID+25, @SGGUID+56), (@SGGUID+25, @SGGUID+57), (@SGGUID+25, @SGGUID+58), (@SGGUID+25, @SGGUID+59), (@SGGUID+25, @SGGUID+60), (@SGGUID+25, @SGGUID+61);
```
Comment (miraco) [inline Updates/Instances/532_karazhan.sql:3334]: This is also not working atm, due to wrong implementation for spawn_group_linked_group
spawn_group_linked aggro only gets trigert when the other group (in this case @sgguid+25) has flag 
CREATURE_GROUP_EVENT_AGGRO
but this group is only the 1 boss mob. Adding CREATURE_GROUP_EVENT_AGGRO to sgguid+25 will crash the server

**Comment by insunaa** [comment]
The trash around Midnight/Attumen should *not* aggro on boss pull. In fact it's a Karazhan speedrun strategy to pop goblin rocket boots and invisibility potion to skip all trash until Midnight

**Comment by miraco** [comment]
[Spectral Stable Hands](https://wowpedia.fandom.com/wiki/Spectral_Stable_Hand) should be killed first due to their heal spell. [[Shackle Undead]](https://wowpedia.fandom.com/wiki/Shackle_Undead) and [[Freezing Trap]](https://wowpedia.fandom.com/wiki/Freezing_Trap) may be used to control other adds. If trash has respawned or has been skipped, those nearby (up to approximately 25 mobs) will join the fight when Midnight is aggro'd.

https://wowpedia.fandom.com/wiki/Attumen_the_Huntsman

**Comment by AnonXS** [comment]
`1. When players attack midnight, all trashs around him should aggro` can this not be handled in any feasable way?

**Pull request merged**